File: go/validanagram/validanagram_test.go
package validanagram

import (
	"testing"
)

func TestValidAnagram(t *testing.T) {
	cases := []struct {
		input1 string
		input2 string
		want   bool
	}{
		{
			input1: "anagram",
			input2: "nagaram",
			want:   true,
		},
		{
			input1: "rat",
			input2: "car",
			want:   false,
		},
	}

	for _, test := range cases {
		if got := isAnagram(test.input1, test.input2); got != test.want {
			t.Errorf("got %v, want %v, test case: %+v", got, test.want, test)
		}
	}
}

File: ts/src/arrays-and-hashing/valid-anagram.test.ts
import { test, expect } from "vitest";
import isAnagram from "./valid-anagram";

test("valid anagram", () => {
  const testCases: Array<{ s: string; t: string; want: boolean }> = [
    {
      s: "anagram",
      t: "nagaram",
      want: true,
    },
    {
      s: "rat",
      t: "car",
      want: false,
    },
  ];

  for (const testCase of testCases) {
    const got = isAnagram(testCase.s, testCase.t);
    expect(got).toBe(testCase.want);
  }
});
